File: frontend/src/app/types/resume.ts
export interface Experience{
    company: string;
    position: string;
    start_date: string;
    end_date: string | null;
    description: string | string[];
}

export interface Education{
    institution: string
    degree: string;
    field_of_study: string;
    start_date: string;
    end_date: string | null;
}

export interface Certificates{
    title: string;
    institution: string
    date_of_receive: string;
    id: number | null;
    link: string | null;
}

export interface Resume{
    full_name: string;
    email: string;
    phone: string | null;
    summary: string | null;
    certificates: Certificates[]
    experiences: Experience[];
    education: Education[];
}

File: backend/models/resume.py
from pydantic import BaseModel
from typing import Optional, List

class Experience(BaseModel):
    company: str
    position: str
    start_date: str
    end_date: Optional[str] = None
    description: List[str] = []

class Education(BaseModel):
    institution: str
    degree: str
    field_of_study: Optional[str] = None
    start_date: str
    end_date: Optional[str] = None

class Certyficates(BaseModel):
    title: str
    institution: str
    date_of_receive: str
    id: Optional[str] = None
    link: Optional[str] = None

class Resume(BaseModel):
    full_name: str
    email: str
    phone: Optional[str] = None
    summary: Optional[str] = None
    certyficates: Optional[List[Certyficates]] = []
    experiences: List[Experience] = []
    education: List[Education] = []

class TailorRequest(BaseModel):
    base_resume_text: str
    job_description_text: str